hazelcast/src/main/java/com/hazelcast/config/CacheConfig.java

```diff
@@ -597,6 +595,8 @@ public void readData(ObjectDataInput in) throws IOException {
 
         mergePolicy = in.readUTF();
         disablePerEntryInvalidationEvents = in.readBoolean();
+
+        setClassLoader(in.getClassLoader());
```

Comment:
Keep a reference to `SerializationService`'s class loader, so when `keyType`/`valueType` are being resolved, the proper classloader will be used.